### PR TITLE
Feature: Adding order manager and unit test

### PR DESCRIPTION
**Pull request by eapg**
* Adding order manager
* Adding order manager unit test

* Description: The order manager will be use to manage the orders status, and will be implement in the kitche simulator for the chefs to take new orders, change order status to "in process" and at the end put the order in an "complete" status. for the differences status the order manager will used queue(placed_order_queue, in_process_order_queue, complete_order_queue, cancel_order_queue).

availables chef will take new orders from the placed_order_queue, they will change order status to in process putting placed orders in the in_process_order_queue while they work in the orders, when the order is complete they will place the order in the complete_order_queue.

the init_order_storage method will be use to create al the queue that the order manager will manage. example: "placed order", "ready to prepate", "preparing", "ready or complete", "cancel".

### CODE REVIEW

File: src/core/order_manager.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    def __init__(self):
+
+        self._order_storage = {}
+        self.order_id_count = 1
+
+    def init_order_storage(self, order_status):
+
+        for status in order_status:
+
```
Comment (reynoldmorel) [inline src/core/order_manager.py:16]: You can remove empty line here

Comment (reynoldmorel) [reply src/core/order_manager.py:16]: Remove unnecessary empty lines in this file

File: src/tests/core/order_manager_test.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import unittest
+from src.core.order_manager import OrderManager
+from src.tests.utils.fixtures.order_fixture import build_order
+
+
+class OrderManagerTestCase(unittest.TestCase):
+    def setUp(self):
+        self.order_manager = OrderManager()
```
Comment (reynoldmorel) [inline src/tests/core/order_manager_test.py:8]: Inconsistencies when adding empty lines, try to make it more consistent

**Review by reynoldmorel — CHANGES_REQUESTED** [review]
Let's wait until you need this, because could more complex than it actually is

File: src/core/order_manager.py (hunk 1 above)
Comment (eapg) [reply src/core/order_manager.py:13]: done!

File: src/tests/core/order_manager_test.py (hunk 2 above)
Comment (eapg) [reply src/tests/core/order_manager_test.py:8]: done!

**Review by reynoldmorel — CHANGES_REQUESTED** [review]
Since tthis is related to order and status, I think you should pass down the whole order and make the let the `OrderManager` decide what to take from the order, according to you idea.

File: src/core/order_manager.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# order manager mechanism
+
+from queue import Queue
+
+
+class OrderManager:
+    def __init__(self):
+        self._order_storage = {}
+
+    def init_order_storage(self, order_status):
```
[the remark this answers is not in the document]
Comment (eapg) [reply src/core/order_manager.py:10]: done. order_statuses were create inside init_order_storage method.

File: src/core/order_manager.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        for status in order_status:
+
+            self._order_storage[status] = Queue(maxsize=0)
+
```
[the remark this answers is not in the document]
Comment (eapg) [reply src/core/order_manager.py:14]: by default when you set maxsize=0 means that the queue could store infinity values.

File: src/tests/core/order_manager_test.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
```
[the remark this answers is not in the document]
Comment (eapg) [reply src/tests/core/order_manager_test.py:8]: done.

File: src/tests/core/order_manager_test.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        self.order_manager.init_order_storage(["order placed"])
+
+    def test_add_to_order_storage(self):
+        order = build_order()
+
+        self.order_manager.add_to_order_storage(order, "order placed")
+        self.assertEqual(1, order.id)
```
[the remark this answers is not in the document]
Comment (eapg) [reply src/tests/core/order_manager_test.py:15]: this was a mistake.

File: src/lib/entities/order.py [hunk 7]
```diff
@@ -7,6 +7,8 @@ def __init__(self):
 
         self.id = None  # integer
         self.order_details = None  # list
+        self.order_status = None # enum
```
Comment (reynoldmorel) [inline src/lib/entities/order.py:10]: could you rename to `status`? This is already within the `order` entity

File: src/core/order_manager.py [hunk 8]
```diff
@@ -0,0 +1,25 @@
+# order manager mechanism
+
+from queue import Queue
+
+
+class OrderManager:
+    def __init__(self):
+        self._order_storage = {
```
Comment (reynoldmorel) [inline src/core/order_manager.py:8]: Could you rename to `order_status_to_order_queue_map`?

File: src/core/order_manager.py [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    def add_to_order_storage(self, order):
+        self._order_storage[order.order_status.value].put(order)
+
+    def get_from_order_storage(self, order_status):
+        return self._order_storage[order_status].get()
+
+    def get_queue_size(self, order_status):
+        return self._order_storage[order_status].qsize()
+
+    def get_queue_empty(self, order_status):
```
Comment (reynoldmorel) [inline src/core/order_manager.py:24]: Could you rename to `is_order_queue_empty`?

File: src/core/order_manager.py [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+            "order_placed": Queue(maxsize=0),
+            "in_process": Queue(maxsize=0),
+            "cancelled": Queue(maxsize=0),
+            "completed": Queue(maxsize=0),
+        }
+
+    def add_to_order_storage(self, order):
+        self._order_storage[order.order_status.value].put(order)
+
+    def get_from_order_storage(self, order_status):
```
Comment (reynoldmorel) [inline src/core/order_manager.py:18]: Could you rename to `get_queue_from_status`?

File: src/core/order_manager.py [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+class OrderManager:
+    def __init__(self):
+        self._order_storage = {
+            "order_placed": Queue(maxsize=0),
+            "in_process": Queue(maxsize=0),
+            "cancelled": Queue(maxsize=0),
+            "completed": Queue(maxsize=0),
+        }
+
+    def add_to_order_storage(self, order):
```
Comment (reynoldmorel) [inline src/core/order_manager.py:15]: Could you rename to `add_to_queue`?

File: src/lib/entities/order.py (hunk 7 above)
Comment (reynoldmorel) [reply src/lib/entities/order.py:10]: Or maybe `preparation_status`? up to you

**Review by reynoldmorel — CHANGES_REQUESTED** [review]

File: src/lib/entities/order.py (hunk 7 above)
Comment (eapg) [reply src/lib/entities/order.py:10]: done. renamed to status.

File: src/core/order_manager.py (hunk 8 above)
Comment (eapg) [reply src/core/order_manager.py:8]: done

File: src/core/order_manager.py (hunk 9 above)
Comment (eapg) [reply src/core/order_manager.py:24]: done.

File: src/core/order_manager.py (hunk 10 above)
Comment (eapg) [reply src/core/order_manager.py:18]: done

File: src/core/order_manager.py (hunk 11 above)
Comment (eapg) [reply src/core/order_manager.py:15]: done

**Review by reynoldmorel — APPROVED** [review]

**Review by reynoldmorel — APPROVED** [review]

**Pull request merged**